web/yo/app/scripts/controllers/reportgenerator.js

```diff
@@ -295,14 +295,55 @@ angular.module('oncokbApp')
 
                 DatabaseConnector.getReviewedData($scope.reviewedData[$scope.evidenceType].evidenceTypes).then(function(response) {
                     if ($scope.evidenceType === 'geneType') {
-                        _.each(response, function(item) {
-                            $scope.reviewedData.geneType.body.push({
-                                gene: item.hugoSymbol,
-                                oncogene: item.oncogene,
-                                tsg: item.tsg
+                        var variantLookupBody = _.map(response, function(item) {
+                            return {
+                                hugoSymbol: item.hugoSymbol
+                            };
+                        });
+                        var geneWithVariants = [];
+                        DatabaseConnector.lookupVariants(variantLookupBody).then(function(result) {
+                            _.each(result, function(items) {
+                                var tempObj = {};
+                                _.each(items, function(item) {
+                                    if (_.isEmpty(tempObj)) {
+                                        tempObj = {
+                                            gene: item.gene.hugoSymbol,
+                                            oncogene: item.gene.oncogene,
+                                            tsg: item.gene.tsg,
+                                            truncatingMutations: false,
+                                            deletion: false,
+                                            amplification: false
+                                        };
+                                        geneWithVariants.push(item.gene.hugoSymbol);
+                                    }
+                                    if (item.alteration === 'Truncating Mutations') {
+                                        tempObj.truncatingMutations = true;
+                                    }
+                                    if (item.alteration === 'Deletion') {
+                                        tempObj.deletion = true;
+                                    }
+                                    if (item.alteration === 'Amplification') {
+                                        tempObj.amplification = true;
+                                    }
+                                });
+                                if (!_.isEmpty(tempObj)) {
+                                    $scope.reviewedData.geneType.body.push(tempObj);
+                                }
                             });
+                            _.each(response, function(item) {
+                                if (geneWithVariants.indexOf(item.hugoSymbol) === -1) {
```

Comment:
If this is the only place using `geneWithVariants`, suggest changing to object implementation for better performance. And this should be used for similar purpose across the project.

Reply:
Good point!